subsys/net/ip/tcp2.c

```diff
@@ -376,10 +376,26 @@ static const char *tcp_conn_state(struct tcp *conn, struct net_pkt *pkt)
 	return buf;
 }
 
-static bool tcp_options_check(void *buf, ssize_t len)
+static u8_t *tcp_options_get(struct net_pkt *pkt, int tcp_options_len)
+{
+	static u8_t options[40]; /* TCP header max options size is 40 */
+	struct net_pkt_cursor backup;
+
+	net_pkt_cursor_backup(pkt, &backup);
+	net_pkt_cursor_init(pkt);
+	net_pkt_skip(pkt, net_pkt_ip_hdr_len(pkt) + net_pkt_ip_opts_len(pkt) +
```

Comment:
Can you check return values of net_pkt_xxx() api's and proceed further? (e.g. what happens if skip is failed)

Reply:
Good point, let's leave that for refactoring since also in other places of TCP2 the return values are not checked.

Reply:
ok